### ISSUE TITLE
Test additional BaseObject functionality

### ISSUE THREAD

**Issue by RNKuhns** [sktime/skbase#63]
**Is your feature request related to a problem? Please describe.**
We test the primary interfaces provided by `BaseObject` but we still need to test some additional functionality. Running list is provided below.

**Describe the solution you'd like**

We need to add tests of the following:
- `BaseObject._get_init_signature()` is `object.__init__`
- `BaseObject_get_init_signature()` raises error for `VAR_POSITIONAL`
- `BaseObject.get_param_names()`
- `BaseObject.set_params()` with no params returns original object.
- `BaseObject.set_params()` raises error when param is not a valid params.
- `BaseObject.set_tags()` tag setting when underlying object doesn't have _tags_dynamic attribute (do we want to add it like we currently do (then we add test class) or do we want this to raise error (since this would imply the subclass doesn't correctly inherit the entire`BaseObejct` interface)?
- `BaseObject.clone_tags()`
- `BaseObject.get_test_params()`
- `BaseObejct.create_test_instance()`:
- `BaseObject.create_test_instances_and_names()`
- `BaseObject._has_implementation_of()`
-  `BaseObject._components()` needs to test correct error raised if base_class is not a class or it is a class that isn't a subclass of `BaseObject`.s returns original object.

**Describe alternatives you've considered**
Add tests to covering these areas to `test_base.py`.

**Comment by fkiraly** [comment]
Comments:

* I would only test the public interfaces
* having said that, perhaps `_get_init_signature` and `_has_implementation` should be public?

> `BaseObject.set_tags()` tag setting when underlying object doesn't have _tags_dynamic attribute (do we want to add it like we currently do (then we add test class) or do we want this to raise error (since this would imply the subclass doesn't correctly inherit the entire`BaseObejct` interface)?

Currently, the `_tags_dynamic` attribute is called in `__init__`, but I see that that could be accidentally skipped if, say, a sub-classes `__init__` does not call `super.init`.

However, the test `test_create_test_instance` checks whether `_tags_dynamic` are present, so that should be sufficient to guarantee this?

> `BaseObject.set_params()` with no params returns original object.

Would that not already be tested by `test_set_params` if the empty param set is a test param set?

> `BaseObject.get_test_params()`

Such a test is present here, I wanted to add it:
https://github.com/sktime/sktime/pull/2862

The problem is that I asked it to check for at least two parameter sets present, which most estimators failed.

Perhaps we can split up #2862 and move some parts here (as well as in sktime, temporarily)?

**Comment by fkiraly** [comment]
> `BaseObejct.create_test_instance()`:
> `BaseObject.create_test_instances_and_names()`

These are already tested, no?
By `test_create_test_instance` and `test_create_test_instance_and_names`, first thing that is tested.

**Comment by fkiraly** [comment]
> `BaseObject._components()` needs to test correct error raised if base_class is not a class or it is a class that isn't a subclass of `BaseObject`.s returns original object.

I don't get this one - wouldn't the object in question simply not have the `_component` method?
Could you rephrase the exact condition?
